mailing_list/admin.py

```diff
@@ -43,3 +53,62 @@ def sync_mailinglist_stats(self, request):
 
     def has_add_permission(self, request):
         return False
+
+
+class SubscribesCSVForm(forms.Form):
+    csv_file = forms.FileField()
+
+
+@admin.register(SubscriptionData)
+class SubscriptionDataAdmin(admin.ModelAdmin):
+    list_display = ["subscription_dt", "email"]
+    search_fields = ["email"]
+    change_list_template = "admin/mailinglist_change_list.html"
+
+    email_regex = re.compile("([a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,})")
+
+    def get_urls(self):
+        return [
+            path("import-csv", self.import_csv, name="import_csv")
+        ] + super().get_urls()
+
+    def parse_rows(self, reader):
+        for row in reader:
+            date_str = " ".join(row[0:4])
+            try:
+                dt = datetime.strptime(date_str, "%b %d %H:%M:%S %Y")
+            except ValueError:
+                logger.error(f"Error parsing date {date_str} from {row=}")
+                dt = None
+            # re-merge, the email address isn't always in a consistent position
+            email_matches = re.search(self.email_regex, " ".join(row[6:]))
+            email = email_matches.group(0) if email_matches else None
+            entry_type = row[6]
+            # only save confirmed subscriber entries, it's all we need for now
+            if entry_type != "new":
+                continue
+            if not email:
+                logger.error(
+                    f"Invalid email {row=} {email_matches=} {' '.join(row[6:])=}"
+                )
+                continue
+            yield SubscriptionData(
+                email=email,
+                entry_type=entry_type,
+                list=row[5][:-1],  # strips trailing :
```

Comment:
Could we be more explicit?

```suggestion
                list=row[5].rstrip(":")
```